File: week11/15/1_Mon_palindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:                
        s = s.lower().strip()
                     
        s_ = str()
        for i in range(len(s)):
            if s[i].isalnum():
                s_ = s_ + s[i]       
        
        for i in range(len(s_)//2):
            if s_[i] != s_[-i-1]:
                return False
        return True

File: week11/15/1_Mon_palindrome.c
#include <stdio.h>
#include <stdbool.h>
#include <ctype.h>
#include <string.h>

bool isPalindrome(char *s) {
    char string[1000];
    int j = 0;

    for (int i = 0; s[i] != '\0'; i++) {
        if (isalnum(s[i])) {
            string[j++] = tolower(s[i]);
        }
    }
    string[j] = '\0';

    int len = strlen(string);
    for (int i = 0; i < len / 2; i++) {
        if (string[i] != string[len - i - 1]) {
            return false;
        }
    }
    return true;
}
